File: steering-hub-backend/steering-service/src/main/java/com/steeringhub/steering/service/CategoryNavService.java
package com.steeringhub.steering.service;

import com.steeringhub.steering.dto.response.CategoryNavItem;
import com.steeringhub.steering.dto.response.SteeringNavItem;

import java.util.List;

public interface CategoryNavService {

    List<CategoryNavItem> listCategories(Long parentId);

    List<SteeringNavItem> listSteerings(Long categoryId, int limit);

    void addHierarchy(Long parentId, Long childId, int sortOrder);

    void removeHierarchy(Long parentId, Long childId);
}

File: steering-hub-frontend/src/services/categoryNavService.ts
import { get, post } from '../utils/request';
import type { CategoryNavItem, SteeringNavItem } from '../types';

export const categoryNavService = {
  listCategories: (parentId?: number) => {
    const params = new URLSearchParams();
    if (parentId != null && parentId > 0) params.set('parentId', String(parentId));
    const qs = params.toString();
    return get<CategoryNavItem[]>(`/api/v1/mcp/categories${qs ? `?${qs}` : ''}`).then((r) => r.data);
  },

  listSteerings: (categoryId: number, limit = 50) =>
    get<SteeringNavItem[]>(`/api/v1/mcp/steerings?categoryId=${categoryId}&limit=${limit}`).then((r) => r.data),

  addHierarchy: (parentCategoryId: number, childCategoryId: number, sortOrder = 0) =>
    post('/api/v1/web/category-hierarchy', { parentCategoryId, childCategoryId, sortOrder }),

  removeHierarchy: (parentCategoryId: number, childCategoryId: number) =>
    post('/api/v1/web/category-hierarchy/remove', { parentCategoryId, childCategoryId }),

  createCategory: (data: { name: string; code?: string; description?: string; parentId?: number }) =>
    post('/api/v1/web/categories', data),
};
